### PR TITLE
[IGA] Lagrange Multiplier based Support

### PR DESCRIPTION
**Pull request by tteschemacher**
**Description**
This PR adds the Lagrange Multiplier support implemented by @rickyaristio.

**Changelog**
- Lagrange Multiplier Support
- VECTOR_LAGRANGE_MULTIPLIER_REACTION variable at the IgaApplication

### CODE REVIEW

File: applications/IgaApplication/custom_conditions/lagrange_support_condition.cpp [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+                    const array_1d<double, 3>& displacement = (Has(DISPLACEMENT))
+                        ? this->GetValue(DISPLACEMENT)
+                        : ZeroVector(3);
+
+                    IndexType counter = 0;
+                    for (IndexType i = 0; i < number_of_nodes; i++)
+                    {
+                        for (IndexType n = 0; n < r_N.size1(); ++n) {
+                            if (r_N(n, i) > shape_function_tolerance) {
+                                const array_1d<double, 3> disp = r_geometry[i].FastGetSolutionStepValue(DISPLACEMENT);
```
Comment (philbucher) [inline applications/IgaApplication/custom_conditions/lagrange_support_condition.cpp]:
```suggestion
                                const array_1d<double, 3>& disp = r_geometry[i].FastGetSolutionStepValue(DISPLACEMENT);

```

**Review by philbucher — COMMENTED** [review]
ok but tests are missing ;)

File: applications/IgaApplication/custom_conditions/lagrange_support_condition.cpp [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+                                u[index + 2] = (disp[2] - displacement[2]);
+                                counter++;
+                            }
+                        }
+                    }
+                    for (IndexType i = 0; i < number_of_nodes; i++)
+                    {
+                        for (IndexType n = 0; n < r_N.size1(); ++n) {
+                            if (r_N(n, i) > shape_function_tolerance) {
+                                const array_1d<double, 3>& l_m = r_geometry[i].FastGetSolutionStepValue(VECTOR_LAGRANGE_MULTIPLIER_REACTION);
```
Comment (tteschemacher) [inline applications/IgaApplication/custom_conditions/lagrange_support_condition.cpp]: @rickyaristio the variable is introduced only to enable the computation with the reactions. At this position the core variable VECTOR_LAGRANGE_MULTIPLIER makes more sense.

Comment (rickyaristio) [reply applications/IgaApplication/custom_conditions/lagrange_support_condition.cpp]: Okay, that was a misunderstanding.

**Comment by loumalouomega** [comment]
@tteschemacher did you try to use the Builder and Solver with LM instead of implementing this directly?

**Comment by tteschemacher** [comment]
> @tteschemacher did you try to use the Builder and Solver with LM instead of implementing this directly?

Actually no. Can you help me to work this out?

**Comment by tteschemacher** [comment]
I'm not sure though if this would be something useful, here. Because this condition is able to deal with it without touching any builder and solver.

**Comment by loumalouomega** [comment]
> I'm not sure though if this would be something useful, here. Because this condition is able to deal with it without touching any builder and solver.

Sorry I confused this with a constraint. Sorry for the noise

**Comment by tteschemacher** [comment]
are we good to go here? @dagmawibekel @rickyaristio ?

**Review by dagmawibekel — DISMISSED** [review]

**Comment by tteschemacher** [comment]
thanks @dagmawibekel, there was a conflict with  the includes in the application could you approve it again ;)

**Review by dagmawibekel — APPROVED** [review]

**Pull request merged**